store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/getelements/impl/ParquetElementRetriever.java

```diff
@@ -0,0 +1,322 @@
+                    this.currentPath = this.paths.next();
+                    this.fileIterator = new ParquetFileIterator(this.currentPath, this.fs);
+                    return openParquetReader();
+                }
+            }
+            return null;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (this.currentElement == null) {
+                try {
+                    this.currentElement = next();
+                } catch (NoSuchElementException e) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+
+        @Override
+        public Element next() throws NoSuchElementException {
+            Element e = getNextElement();
+            if (this.needsValidation) {
+                final ElementFilter preAggFilter = view.getElement(e.getGroup()).getPreAggregationFilter();
+                if (preAggFilter != null) {
+                    while (!preAggFilter.test(e)) {
+                        e = getNextElement();
+                    }
+                }
+            }
+            return e;
+        }
+
+        private Element getNextElement() {
+            Element e;
+            try {
+                if (this.currentElement != null) {
+                    LOGGER.debug("Current element: " + this.currentElement);
+                    e = this.currentElement;
+                    this.currentElement = null;
+                } else {
+                    if (this.reader != null) {
+                        GenericRecord record = this.reader.read();
+                        if (record != null) {
+                            e = convertGenericRecordToElement(record);
+                        } else {
+                            LOGGER.debug("Closing Parquet reader");
+                            this.reader.close();
+                            this.reader = openParquetReader();
+                            if (this.reader != null) {
+                                record = this.reader.read();
+                                if (record != null) {
+                                    e = convertGenericRecordToElement(record);
+                                } else {
+                                    LOGGER.debug("This file has no data");
+                                    e = next();
+                                }
+                            } else {
+                                LOGGER.debug("Reached the end of all the files of data");
+                                throw new NoSuchElementException();
+                            }
+                        }
+                    } else {
+                        throw new NoSuchElementException();
+                    }
+                }
+            } catch (IOException | OperationException ex) {
+                throw new NoSuchElementException();
+            }
+            if (e instanceof Edge && this.currentPath.toString().contains("reverseEdges")) {
```

Comment:
This if statement has to come at the end as it applies to the element set within the previous if statement.